### DISCUSSION TITLE
figures and caching

### DISCUSSION THREAD

**Discussion by shabbychef** [quarto-dev/quarto-cli#542] [Q&A] [answered]
I am building an HTML doc, and am using caching to speed up computation. It seems that when I tweak some of the text and rebuild the document, the figures have been deleted and are not remade. For doc `autorisk.qmd`, it is looking for, and not finding, `autorisk_files/figure-html/foo-1.png`. That directory `autorisk_files/figure-html` is built and populated on the first run, but on a re-run it is apparently deleted first.

Not sure it is relevant, but I am using option `self-contained: true`.

**Comment by shabbychef** [comment]
(for now turning off caching in all figure chunks solves my immediate problem.)

**Comment by jjallaire** [comment]
The problem may be that we only prevent deletion of _files if the global `cache` option is set (as opposed to the option for individual chunks). Ideally we could be smart and not delete it in the case of chunk level cache directives. I'll convert this to an issue to track making sure that we do this.

**Reply by shabbychef** [reply]
FWIW, the global `cache` option _is_ set in this document. There are no chunk level cache directives. (well, I added some as a workaround to this very problem, but ideally there would be none.)

**Comment by jjallaire** [comment]
Interesting. `self-contained: true` _should_ be arranging to  leave the `_files` dir alone. I know it did this at one time but perhaps this has regressed? I'll take a look. In the meantime if you set `self-contained: false` the cache should work as expected.

**Comment by jjallaire** [comment]
Sadly it looks like this is also happening w/ `self-contained: false`. Will get to the bottom of it soon!

**Comment by jjallaire** [comment, marked as answer]
Okay, this is now fixed here https://github.com/quarto-dev/quarto-cli/commits/main

Fix should be in tomorrow's build.
